kivy/uix/image.py

```diff
@@ -188,6 +199,16 @@ def get_image_ratio(self):
     defaults to 0.
     '''
 
+    auto_anim_delay = BooleanProperty(False)
+    '''If this property is set True and image is GIF animation,
+    delay of each frame would be get from GIF-image file.
+    
+    .. versionadded:: 1.6.0
```

Comment:
2.0.0

Reply:
Why do we need to define both `auto_anim_delay` and `durations`? Basically, if `durations` is truthy, that's used, otherwise it uses `anim_delay`.

Reply:
> Why do we need to define both `auto_anim_delay` and `durations`? Basically, if `durations` is truthy, that's used, otherwise it uses `anim_delay`.

Compatibility. It must be auto_anim_delay == False and durations is None.

Reply:
But why? By default `durations` is None so the old behavior is used. It's only if `durations` is set that the new behavior is used. Unless I'm reading the code wrong.

Reply:
> But why? By default `durations` is None so the old behavior is used. It's only if `durations` is set that the new behavior is used. Unless I'm reading the code wrong.

I tried to answer below